src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -208,14 +208,14 @@ async Task EnsureLong()
                     case OperationCanceledException _:
                         Debug.Assert(async);
                         e = new TimeoutException("Timeout during reading attempt");
-                        dontBreak = dontBreakOnTimeouts;
+                        dontBreak = true;
```

Comment:
Maybe it's better to just throw the exception here, rather then setting a flag and flowing to the code below. This way the timeouts just throw, whereas any other exception breaks and throws.

Another thought is to merge the async and sync switch cases (two labels, one block), and to just put a ternary inside the Debug.Assert. We can do the same on the writing side as well.